src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private int currentCapacity;
+    private int currentLoadFactor;
+    private Node<K, V>[] storage;
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        currentLoadFactor = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        storage = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeed();
+        int position = getPosition(key);
+        Node<K, V> oldNode = storage[position];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (oldNode == null) {
+            storage[position] = newNode;
+            size++;
+        } else {
+            Node<K, V> nextNode = oldNode;
+            while (nextNode != null) {
+                if (Objects.equals(nextNode.key, newNode.key)) {
+                    nextNode.value = newNode.value;
+                    break;
+                }
+                if (nextNode.nextNode == null) {
+                    nextNode.nextNode = newNode;
+                    size++;
+                    break;
+                } else {
+                    nextNode = nextNode.nextNode;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> resultNode = new Node<>(key, null, null);
+        Node<K, V> currentNode = storage[getPosition(key)];
+        if (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                resultNode = currentNode;
+            } else {
+                while (currentNode != null) {
+                    if (Objects.equals(currentNode.key, key)) {
+                        resultNode = currentNode;
+                        break;
+                    }
+                    currentNode = currentNode.nextNode;
+                }
+            }
+        }
+        return resultNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getPosition(K key) {
+        int position = key == null ? 0 : key.hashCode() % currentCapacity;
+        return position >= 0 ? position : position * -1;
+    }
+
+    private void resizeIfNeed() {
+        if (currentLoadFactor == size) {
+            currentCapacity = currentCapacity << 1;
+            currentLoadFactor = (int) (currentCapacity * DEFAULT_LOAD_FACTOR);
+            Node<K, V>[] oldStorage = storage;
+            storage = (Node<K, V>[]) new Node[currentCapacity];
+            size = 0;
+            for (Node<K, V> node : oldStorage) {
+                if (node != null) {
+                    Node<K, V> nextNode = node;
+                    while (nextNode != null) {
+                        put(nextNode.key, nextNode.value);
+                        nextNode = nextNode.nextNode;
```

Comment:
You are iterating every "bucket", so while cycle will not break the logic without if.

Reply:
nextNode = nextNode.nextNode will break this loop

Reply:
Maybe I don't get something)
You have for:each before while.
So your goal is every while to break, isn't it?